### PR TITLE
Kill Bank::copy_for_tpu in favor of Bank::new_from_parent

### PR DESCRIPTION
**Pull request by aeyakovenko**
#### Problem

copy_for_tpu is not a real fork and just clones the bank.  

#### Summary of Changes

Use Bank::new_from_parent

Fixes #

### CODE REVIEW

**Comment by garious** [comment]
Can you rebase?

File: src/fullnode.rs [hunk 1]
```diff
@@ -363,9 +363,10 @@ impl Fullnode {
                 }
                 None => FullnodeReturnType::LeaderToLeaderRotation, // value doesn't matter here...
             };
-
+            let mut new: Arc<Bank> = Arc::new(Bank::new_from_parent(self.bank.clone()));
```
Comment (garious) [inline src/fullnode.rs]: How about:

```rust
let tpu_bank = self.bank().new_from_parent();
```

File: src/fullnode.rs [hunk 2]
```diff
@@ -363,9 +363,10 @@ impl Fullnode {
...
+            std::mem::swap(&mut self.bank, &mut new);
```
Comment (garious) [inline src/fullnode.rs]: Too cute

Comment (aeyakovenko) [reply src/fullnode.rs]: @garious the problem is when do we get that bank back and assign it to self, which is what the tvu uses.

i don't see a good synchronization point.

Comment (aeyakovenko) [reply src/fullnode.rs]: nm, it works!

File: src/fullnode.rs [hunk 3]
```diff
@@ -363,10 +363,9 @@ impl Fullnode {
                 }
                 None => FullnodeReturnType::LeaderToLeaderRotation, // value doesn't matter here...
             };
-            let mut new: Arc<Bank> = Arc::new(Bank::new_from_parent(self.bank.clone()));
-            std::mem::swap(&mut self.bank, &mut new);
+            let tpu_bank = Arc::new(Bank::new_from_parent(self.bank.clone()));
```
Comment (garious) [inline src/fullnode.rs]: Until the bank traverses parents to implement all its methods, that clone() should be within `new_from_parent()`.

**Comment by garious** [comment]
ha, that's not was I was going for. I mean you need to clone the underlying bank, not the reference to it

**Comment by garious** [comment]
Nevermind. I see you already added all the smarts to traverse the parents. This looks good.

**Pull request merged**